sw/device/silicon_creator/lib/cert/cbor.c

```diff
@@ -0,0 +1,146 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/cert/cbor.h"
+
+#include <string.h>
+
+#include "include/dice/cbor_writer.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/silicon_creator/lib/error.h"
+
+#define CBOR_RETURN_IF_OVERFLOWED(p)    \
+  do {                                  \
+    if (CborOutOverflowed(p)) {         \
+      LOG_ERROR("CborOutOverflowed!!"); \
+      return kErrorCertInvalidSize;     \
+    }                                   \
+  } while (0)
+
+#define CBOR_CHECK_OVERFLOWED_AND_RETURN(p) \
+  do {                                      \
+    CBOR_RETURN_IF_OVERFLOWED(p);           \
+    return kErrorOk;                        \
+  } while (0)
+
+inline rom_error_t cbor_write_out_init(struct CborOut *p, void *buf,
```

Comment:
I think you want to keep all of these inline functions in the header file and mark then `static inline`. You can still share them with everything that include the header. It ensure they get inlined without LTO: https://stackoverflow.com/a/47821267 (applies to all inline functions)